test/remote.spec.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+const Cache = require('../lib/cache');
+const config = require('../lib/config');
+const sinon = require('sinon');
+const path = require('path');
+const fs = require('fs-extra');
+const index = require('../lib/index');
+const utils = require('../lib/utils');
+var assert = require('assert');
+
+describe('Remote', () => {
+  describe('update()', () => {
+    const TIMEOUT_INTERVAL = 60000; // 1min timeout for each test case.
+    let tempFolder;
+    let i = 0;
+
+    const testCases = [
+      {
+        description: 'No language specified',
+        LANG: ['en'],
+        expectedFolders: ['pages'],
+      },
+      {
+        description: '1 Language Specified that doesn\'t exist',
+        LANG: ['pt_BB'],
+        expectedFolders: ['pages'],
+      },
+      {
+        description: '1 Language Specified that does exist',
+        LANG: ['ca'],
+        expectedFolders: ['pages', 'pages.ca'],
+      },
+      {
+        description: 'Languages Specified that exist',
+        LANG: ['pt', 'pt_BR'],
+        expectedFolders: ['pages', 'pages.pt_BR'],
+      },
+      {
+        description: 'Multiple Languages Specified that exist',
+        LANG: ['pt_BR', 'pt', 'en', 'hi', 'mo'],
+        expectedFolders: ['pages', 'pages.hi', 'pages.pt_BR'],
+      },
+    ];
+
+    beforeEach(() => {
+      // Iterate over the test cases
+      sinon.spy(fs, 'ensureDir');
+      sinon.stub(fs, 'remove').resolves();
+      sinon.stub(fs, 'copy').resolves();
+      sinon.stub(utils, 'localeToLang').returns(testCases[i].LANG);
```

Comment:
Instead of keeping track of an index variable to use in a `beforeEach`, it's more straightforward I think to do something like this:

```
testCases.forEach((testCase) => {
  describe(testCase.description, () => {
    before(() => {
      // set up sinon
    });
    it('passes', () => {
      // test
    });
    after(() => {
      // break down sinon
    });
  });
});
```

Reply:
Thank you for this. It was my first time using mocha so I was struggling with this part a bit.